### PR TITLE
Attachments total amount and size

### LINKED ISSUE #4385: total attachment size reset
The total attachment size var in smf_fileUpload.js doesn't get reseted when a file is canceled

Steps to reproduce:

- Set a total attachment size limit, SMF by default uses 192KB.
- Upload a file bigger than that size. for example 256KB
- Cancel the file.
- Upload another file. doesn't matter which size, as an example, 50KB

The old file is still counted towards the total size, the new file will also be added:

256 + 50

The error will display 306KB, doesn't matter if the second file surpasses the limit of not.

I remember I fixed this but looking at the file a few more changes were added so I cannot easily identify where the issue is.

image

image

Edit, added some images to help identify the issue.

### LINKED ISSUE #4386: total attachments amount
The total amount of attachments an user can upload isn't been applied.

By default SMF limits the number of attachments to 4, currently the user can keep on dragging or selecting files with no limit been applied.

The file extension check also isn't been applied, by default SMF doesn't allow .php files but I could keep added them.

image

image

### PR DESCRIPTION
**Pull request by MissAllSunday**
fixes #4386
fixes #4385

Adds back the events for canceled and maxfilesexceeded
Updates Dropzone to 5.2.0

### CODE REVIEW

**Comment by MissAllSunday** [comment]
It seems I borked Scrutinizer...  110 minutes performing "PHP Analyzer: Running Analysis" and counting...

**Comment by frandominguezl** [comment]
+14h running now xD

**Comment by MissAllSunday** [comment]
OK that didn't work...

**Comment by colinschoen** [comment]
Will review soon.

**Comment by colinschoen** [comment]
I am still able to upload more files than the max amount. Is this working for you locally?

**Comment by MissAllSunday** [comment]
did a couple of tests but will check it again.

**Comment by colinschoen** [comment]
Bump... :)

**Review by colinschoen — CHANGES_REQUESTED** [review]
Marking as needs attention so this isn't still pending my review. PTAL when able.

**Comment by MissAllSunday** [comment]
sorry, my PC died out and I haven't had the time to setup a proper dev env.

**Comment by colinschoen** [comment]
No worries. Any chance you will have time to work on this in the near future?

**Comment by MissAllSunday** [comment]
OK, I believe I got all the checks covered, theres still the multi-file select issue but I'm afraid thats a limitation of the plugin itself and haven't had found a way around it,

File: Themes/default/scripts/smf_fileUpload.js [hunk 1]
```diff
@@ -74,18 +63,25 @@ function smf_fileUpload(oOptions) {
 		},
 		accept: function (file, done) {
 
+			var currentlyUsedKB = myDropzone.options.currentUsedSize / 1024,
+				totalKB = myDropzone.options.maxTotalSize,
+				fileKB = myDropzone.options.maxFileSize,
+				uploadedFileKB = file.size / 1024;
+
+			// Check againts the max amount of files setting
```
Comment (colinschoen) [inline Themes/default/scripts/smf_fileUpload.js]: nit: againts -> against

**Review by colinschoen — APPROVED** [review]
LGTM nice work

1 spelling nit

**Pull request merged**